plugin/handler_socket/libhsclient/allocator.hpp

```diff
@@ -42,6 +43,19 @@ typedef std::allocator<int> allocator_type;
 	static_cast<typ *>(malloc((len) * sizeof(typ)))
 #define DENA_ALLOCA_FREE(x) free(x)
 #endif
+*/
+
+// AIX
+#if 1
+#define DENA_ALLOCA_ALLOCATE(typ, len) \
+        static_cast<typ *>((void*)alloca((len) * sizeof(typ)))
+#define DENA_ALLOCA_FREE(x)
+#else
+#define DENA_ALLOCA_ALLOCATE(typ, len) \
+        static_cast<typ *>((void*)malloc((len) * sizeof(typ)))
+#define DENA_ALLOCA_FREE(x) free(x)
+#endif
+
```

Comment:
Please see the above hunk: it clearly comments out code and then adds a copy again. Similarly in spider/hs_client.

Do you have any reference to why alloca() return char* on AIX?